engine-plugins/identity-ldap/src/main/java/org/camunda/bpm/identity/impl/ldap/LdapGroupQuery.java

```diff
@@ -40,6 +44,14 @@ public LdapGroupQuery(CommandExecutor commandExecutor) {
     super(commandExecutor);
   }
 
+  // overrides
+
+  public GroupQuery groupNameLike(String nameLike) {
+    super.groupNameLike(nameLike);
+    this.nameLike = this.nameLike.replaceAll(DB_QUERY_WILDCARD, LDAP_QUERY_WILDCARD);
```

Comment:
Instead of converting the `%` inside the API method calls, do you see advantages of converting them in `LdapIdentityProviderSession`?

Reply:
No, I don't see any advantages of doing it in the `LdapIdentityProviderSession`. Do you?
I chose to implement it in the API calls because:
* It creates valid queries as early as possible. Since queries that use the correct wildcard from the beginning work too, I think it is nice to convert as early as possible so the behavior is consistent after the query object was created.
* I find it more intuitive to adjust the API methods because I believe it is easier to understand for users who want to implement something similar.

Reply:
I think all LDAP search filter conversions should be made at a single and central place. As a developer, who has to adjust the logic in the future, I can easily find all search filter conversions at one glance instead of debugging multiple places to find out where similar things happen. We should consolidate similar logic rather than spread them in different places. Also, I think this pays into the separation of concerns principle.

Regarding your arguments:

> It creates valid queries as early as possible

Is there are concrete use case for this? What does it allow a user to do?

> more intuitive to adjust the API methods

I think we should mainly strive to make our implementation maintainable for ourselves.

Reply:
> I think all LDAP search filter conversions should be made at a single and central place.

I thought of it more like this:
The API calls produce a query object. The `LdapIdentityProviderSession` converts this query to a filter that is understood by LDAP and executes it. In that sense, my implementation respects the separation of concerns because the conversion from query to filter does not have to do anything with the conversion of the wildcard character.

>Is there are concrete use case for this? What does it allow a user to do?
>>It creates valid queries as early as possible

There is no use case for the end user. It's more about consistency. The LDAP query object using the DB wildcard character is invalid since it does not work with LDAP. I believe it is better to create a valid query object than to fix it later when the actual request is made.

In the end, it's your decision. I am happy to move the implementation if you think we should do it.

Reply:
Decision: Implementation moves to `LdapIdentityProviderSession`